async-clipboard/demo.js

```diff
@@ -0,0 +1,55 @@
+const copy = document.querySelector('#copy');
+const paste = document.querySelector('#paste');
+const out = document.querySelector('#out');
+
+/** Write contents of the textarea to the clipboard when clicking "Copy" */
+copy.onclick = () => {
+  navigator.clipboard.writeText(out.value)
+    .then(result => {
+      ChromeSamples.log('Text copied.' + result);
+    })
+    .catch(err => {
+      ChromeSamples.log('Failed to copy text: ' + err);
+    });
+};
+
+/** Read from clipboard when clicking the Paste button */
+paste.onclick = () => {
+  navigator.clipboard.readText()
+    .then(text => {
+      out.value = text;
+      ChromeSamples.log('Text pasted.');
+    })
+    .catch(err => {
+      ChromeSamples.log('Failed to read from clipboard: ' + err);
+    });
+};
+
+/** Watch for pastes */
+document.addEventListener('paste', e => {
+  e.preventDefault();
+  navigator.clipboard.getText().then(text => {
+    ChromeSamples.log('Updated clipboard contents: ' + text);
+  });
+});
+
+/** Set up buttons to request permissions and display status: */
+document.querySelectorAll('data-permission').forEach(btn => {
+  const permission = btn.getAttribute('data-permission');
+  navigator.permissions.query({name: permission})
```

Comment:
FWIW, I'm still seeing `Uncaught (in promise) TypeError: Failed to read the 'query' property from 'Permissions': Async Clipboard flag is not enabled.` error.
You may want to check if this is not part of what is actually shipping.

I'm on Google Chrome	67.0.3363.0 (Official Build) canary (64-bit)

Reply:
I'm seeing this too. Looks like the permissions portion of the API is still under the experimental flag, so I've added a note and accounted for it in the example code.